web/src/api/auth/login-callback.ts

```diff
@@ -133,6 +150,93 @@ export const auth0Login = withApiAuthRequired(
         invite_id ? urls.signup({ invite_id }) : urls.signup()
       );
     }
+    let invite: InviteQuery["invite"][number] | null = null;
+
+    if (invite_id) {
+      try {
+        const fetchInviteResult = await FetchInviteSdk(client).Invite({
+          id: invite_id,
+        });
+
+        if (!fetchInviteResult) {
+          throw new Error(`Error while fetching invite: ${invite_id}`);
+        }
+
+        if (fetchInviteResult.invite.length > 0) {
+          invite = fetchInviteResult.invite[0];
+        }
+      } catch (error) {
+        logger.error("Error while fetching invite for FetchInviteSdk.", {
+          error,
+        });
+
+        return res.redirect(
+          307,
+          urls.logout({ login_error: LoginErrorCode.Generic })
+        );
+      }
+
+      if (!invite || !invite.team_id) {
+        logger.error("Invite not found or team_id is missing.");
+
+        return res.redirect(
+          307,
+          urls.logout({ login_error: LoginErrorCode.Generic })
+        );
+      }
+
+      let membership: InsertMembershipMutation["insert_membership_one"] | null =
+        null;
+
+      try {
+        const insertMembershipResult = await InsertMembershipSdk(
+          client
+        ).InsertMembership({
+          team_id: invite.team_id,
+          user_id: user.id,
+          role: Role_Enum.Member,
+        });
+
+        membership = insertMembershipResult.insert_membership_one;
+      } catch (error) {
+        logger.error(
+          "Error while inserting membership for InsertMembershipSdk.",
+          {
+            error,
+          }
+        );
+
+        return res.redirect(
+          307,
+          urls.logout({ login_error: LoginErrorCode.Generic })
+        );
+      }
+
+      if (!membership) {
+        logger.error("Membership not found after inserting.");
+
+        return res.redirect(
+          307,
+          urls.logout({ login_error: LoginErrorCode.Generic })
+        );
+      }
+
+      try {
+        const deleteInviteResult = await DeleteInviteSdk(client).DeleteInvite({
+          invite_id,
+        });
+
+        if (!deleteInviteResult.delete_invite_by_pk) {
```

Comment:
given the user is already inserted perhaps best to log an error but let the user continue? invitation would be useless anyways?

Reply:
In this case we need to additionally check that user's data that we can get by invite_id matches with the data from auth0. For example email.
Thanks

Reply:
Btw we are throwing it inside trycatch, so it will be just logged

Reply:
Updated